src/azul/http.py

```diff
@@ -386,3 +386,39 @@ def urlopen(self,
                         return response
             else:
                 return response
+
+
+def parse_header(header: str) -> tuple[str, dict[str, str]]:
+    """
+    Parse representation headers, like content-type, and return a key
+    consisting of the media type (MIME type) and the parameters/directives of
+    the specified media type.
+
+    >>> parse_header('text/html; charset=utf-8')
+    ('text/html', {'charset': 'utf-8'})
+
+    >>> key, params = parse_header('application/json; charset=utf-8; foo=bar')
+    >>> key
+    'application/json'
+    >>> params.items()
+    dict_items([('charset', 'utf-8'), ('foo', 'bar')])
+
+    >>> parse_header('text/html')
+    ('text/html', {})
+
+    >>> parse_header('')
+    ('', {})
+    """
+    from email.message import (
+        Message,
+    )
+    m = Message()
+    m.add_header('content-type', header)
```

Comment:
I an prior review this was a kwarg with a default value, and I objected to the default value because it was arbitrary. But hard-coding this value is the other extreme, and also not what we want.